storage/storage_test.go

```diff
@@ -248,6 +246,8 @@ func makeLayer(t *testing.T, compression archive.Compression) (ddigest.Digest, i
 	for i := 1024; i < 2048; i++ {
 		buf[i] = 0
 	}
+	// SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test
+	errs := make(chan error, 1)
```

Comment:
Don’t comment what the code _was_, say what it does _now_.

Did you mean something like “errs exists to return an error from the goroutine, because the `t.Fatalf` can’t be called by any goroutines newly created by the test”?

In that case I’d instead prefer the goroutine itself to start with a “Warning: don’t call `t.Fatalf` from this goroutine, it must be called from the primary goroutine”, or something like that — that’s what _everyone_ must know. The purpose of `errs` is less important, and follows naturally.

The name `errs` suggests that there could be _multiple_ values returned; that’s not what the intends to implement AFAICS.